catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -205,57 +230,140 @@ public DeleteRequest process(DeleteRequest input)
+      Map<String, String> properties) throws DefaultSecurityAttributeValuesPluginException {
+    final String karafEtcSystemPropertyName = "karaf.etc";
+    final String karafEtcSystemPropertyValue = System.getProperty(karafEtcSystemPropertyName);
+    if (karafEtcSystemPropertyValue == null) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          "Unable to get " + karafEtcSystemPropertyName + " system property value");
+    }
+
+    final String userAttributesFilePath =
+        karafEtcSystemPropertyValue + File.separator + USERS_ATTRIBUTES_FILE_NAME;
+    final JsonElement usersDotAttributesFileJsonElement;
+    try (final Reader reader = new FileReader(userAttributesFilePath)) {
+      usersDotAttributesFileJsonElement = new JsonParser().parse(reader);
+    } catch (IOException | JsonSyntaxException e) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          "Unable to read " + USERS_ATTRIBUTES_FILE_NAME + " file", e);
+    }
+
+    if (!usersDotAttributesFileJsonElement.isJsonObject()) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          UNEXPECTED_FORMAT_IN_USERS_DOT_ATTRIBUTES_MESSAGE
+              + "Expected contents to be a json object");
+    }
+
+    final JsonObject usersDotAttributesFileJsonObject =
+        usersDotAttributesFileJsonElement.getAsJsonObject();
+    final String hostSystemPropertyName = SystemBaseUrl.HOST;
+    final String hostSystemPropertyValue = System.getProperty(hostSystemPropertyName);
+    if (!StringUtils.isNotEmpty(hostSystemPropertyValue)) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          "Unable to get " + hostSystemPropertyName + " system property value");
+    }
+
+    if (!usersDotAttributesFileJsonObject.has(hostSystemPropertyValue)) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          UNEXPECTED_FORMAT_IN_USERS_DOT_ATTRIBUTES_MESSAGE
+              + "Could not find "
+              + hostSystemPropertyValue
+              + " user");
+    }
+
+    final JsonElement systemHighUserJsonElement =
+        usersDotAttributesFileJsonObject.get(hostSystemPropertyValue);
+    if (!systemHighUserJsonElement.isJsonObject()) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          UNEXPECTED_FORMAT_IN_USERS_DOT_ATTRIBUTES_MESSAGE
+              + "Expected a json object for the system high user");
+    }
+
+    final JsonObject systemHighUserJsonObject = systemHighUserJsonElement.getAsJsonObject();
+    if (MapUtils.isEmpty(properties)) {
+      throw new DefaultSecurityAttributeValuesPluginException("Properties may not be empty");
+    }
+
+    final Map<String, List<String>> newMetacardAttributeNameToSystemHighValues = new HashMap<>();
+    for (Map.Entry<String, String> entry :
+        CONFIGURATION_KEY_TO_METACARD_ATTRIBUTE_NAME_MAP.entrySet()) {
+      final String configurationKey = entry.getKey();
+      if (!properties.containsKey(configurationKey)) {
```

Comment:
I think this check could be removed as it will be caught by the next check below.

Reply:
I like the 2 different log messages: one for the missing key and another for the blank value.